backend/src/pg.js

```diff
@@ -0,0 +1,84 @@
+const Url = require('url');
+const Pool = require('pg-pool');
+
+// WITHIN_CONTAINER is set in docker-compose.yml
+// We are *not* running within a local container when we run jest tests, e. g. via `yarn test`
+const isRunningWithinLocalContainer = process.env.WITHIN_CONTAINER === 'true';
+const host = isRunningWithinLocalContainer ? 'db' : 'localhost';
+// user, password, and the database name correspond to those set in docker-compose.yml
+const pgLocalConnectionString = `postgres://postgres:postgres@${host}:5432/postgres`;
+
+// See https://stackoverflow.com/a/28489160
+const isProduction = !!(process.env._ && (process.env._.indexOf('heroku') >= 0));
```

Comment:
Any reason we can't just use `NODE_ENV` to find out if we're running in production?

Reply:
I guess it requires more setup actions by the developers locally (`export NODE_ENV=dev`) and Heroku setup (`heroku config:set NODE_ENV="production"`), imagining some developer wants to test the deployment in their own free Heroku account (like I do right now, but other developers may want to do this, too).

The above line doesn't require anything like that.

Reply:
@leventov Heroku sets that variable by default, and I would assume we never set it locally. It's standard practice to use `NODE_ENV` in NodeJS apps.